components/event-publisher/org.wso2.carbon.event.publisher.core/src/main/java/org/wso2/carbon/event/publisher/core/internal/StringOutputMapper.java

```diff
@@ -0,0 +1,273 @@
+     *
+     * @param mappingText
+     * @return
+     * @throws EventPublisherConfigurationException
+     */
+    protected String validateForType(String mappingText) throws EventPublisherConfigurationException {
+
+        return mappingText; // Do nothing by default
+    }
+
+    /**
+     * Validate the text using validateForType method and then assign to the instance variables.
+     *
+     * @param actualMappingText
+     * @throws EventPublisherConfigurationException
+     * @see StringOutputMapper#validateForType
+     */
+    private void validateAndSetActualMappingText(String actualMappingText) throws EventPublisherConfigurationException {
+
+        // Validate for type specific requirements
+        this.actualMappingText = validateForType(actualMappingText);
+        this.mappingTextList = generateMappingTextList(actualMappingText);
+    }
+
+    /**
+     * Validate the registry path if it is parametrized or the resource mapped by the path.
+     *
+     * @param path
+     * @throws EventPublisherConfigurationException
+     */
+    private void validateRegistryPathWithOutputProperties(String path) throws EventPublisherConfigurationException {
+
+        if (containsParameter(path)) {
+            // Parametrized path. Just verify the path
+            validateStreamDefinitionWithOutputProperties(path);
+            this.customRegistryPath = true;
+        } else {
+            // Not a parametrized path. Get the actual content
+            String actualMappingText = EventPublisherServiceValueHolder.getCarbonEventPublisherService().getRegistryResourceContent(path);
+            validateStreamDefinitionWithOutputProperties(actualMappingText);
+            validateAndSetActualMappingText(actualMappingText);
+        }
+    }
+
+    /**
+     * Validate the given text using the output properties of stream.
+     *
+     * @param text
+     * @throws EventPublisherConfigurationException
+     */
+    private void validateStreamDefinitionWithOutputProperties(String text) throws EventPublisherConfigurationException {
+
+        List<String> mappingProperties = getOutputMappingPropertyList(text);
+        for (String property : mappingProperties) {
+            if (!propertyPositionMap.containsKey(property)) {
+                throw new EventPublisherStreamValidationException("Property " + property + " is not in the input stream definition.",
+                        eventPublisherConfiguration.getFromStreamName() + ":" + eventPublisherConfiguration.getFromStreamVersion());
+            }
+        }
+    }
+
+    /**
+     * Check whether the given text contains atleast one parameter.
+     *
+     * @param text
+     * @return
+     */
+    private boolean containsParameter(String text) {
+
+        return text.contains(EventPublisherConstants.TEMPLATE_EVENT_ATTRIBUTE_PREFIX) && text.indexOf(EventPublisherConstants.TEMPLATE_EVENT_ATTRIBUTE_POSTFIX) > 0;
```

Comment:
Do we really need a separate method for this ? Whether we are calling this from multiple places ?